common/browser_context.go

```diff
@@ -118,6 +118,11 @@ func NewBrowserContext(
 	if opts == nil {
 		opts = DefaultBrowserContextOptions()
 	}
+	// Always use the [Browser]'s user agent if it's not set by the user.
+	// Setting this forces [FrameSession] to set Chromium's user agent.
+	if strings.TrimSpace(opts.UserAgent) == "" {
+		opts.UserAgent = browser.UserAgent()
```

Comment:
The issue describes a proposal where it involves working with two en vars:

```
K6_BROWSER_OMIT_HEADLESS=true
K6_BROWSER_USER_AGENT_EXTRA=Grafana-Synthetic-Monitoring/1.0.0
```

This PR defaults to removing the headless altogether. Is that intended, or are there more PRs coming that will change the behaviour so that it works with some sort of config?

Reply:
Removing `Headless` is our first goal. More PRs are coming soon 👍 Let's first get the original goal out of the way.

Reply:
See: https://github.com/grafana/xk6-browser/issues/1497#issuecomment-2473828191